### PR TITLE
Update DeepSpeed version requirement in Dockerfile

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes PR merge blocking on master.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: dockers/base-cuda/Dockerfile [hunk 1]
```diff
@@ -118,8 +118,7 @@ RUN \
 
 RUN \
     # install DeepSpeed
-    # TODO(@SeanNaren): 0.3.15 is broken - skipping to unblock
-    pip install 'deepspeed>=0.3.14,!=0.3.15'
+    pip install 'deepspeed>=0.3.16
```
Comment (awaelchli) [inline dockers/base-cuda/Dockerfile:121]:
```suggestion
    pip install 'deepspeed==0.3.14
```

**Comment by leezu** [comment]
@awaelchli is there a tracking issue or other resource detailing the problems with deepspeed>0.3.14?

**Comment by awaelchli** [comment]
Yes, a failing test after Deepspeed updated to a new version 
https://dev.azure.com/PytorchLightning/pytorch-lightning/_build/results?buildId=7176&view=logs&j=3afc50db-e620-5b81-6016-870a6976ad29&t=d9f671c5-a304-5675-5394-961fd7f98b9b

```
_______________________ test_deepspeed_multigpu_stage_3 ________________________

tmpdir = local('/tmp/pytest-of-AzDevOps_azpcontainer/pytest-7/test_deepspeed_multigpu_stage_0')
deepspeed_config = {'optimizer': {'params': {'lr': 3e-05}, 'type': 'SGD'}, 'scheduler': {'params': {'last_batch_iteration': -1, 'warmup_max_lr': 3e-05, 'warmup_min_lr': 0, 'warmup_num_steps': 100}, 'type': 'WarmupLR'}}

    @RunIf(min_gpus=2, deepspeed=True, special=True)
    def test_deepspeed_multigpu_stage_3(tmpdir, deepspeed_config):
        """
        Test to ensure ZeRO Stage 3 works with a parallel model.
        """
        model = ModelParallelBoringModel()
        trainer = Trainer(
            plugins=[DeepSpeedPlugin(stage=3)],
            default_root_dir=tmpdir,
            gpus=2,
            fast_dev_run=True,
            precision=16,
        )
>       trainer.fit(model)

tests/plugins/test_deepspeed_plugin.py:459: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pytorch_lightning/trainer/trainer.py:869: in fit
    self._run(model)
pytorch_lightning/trainer/trainer.py:473: in _run
    self.pre_dispatch()
pytorch_lightning/trainer/trainer.py:498: in pre_dispatch
    self.accelerator.pre_dispatch(self)
pytorch_lightning/accelerators/accelerator.py:108: in pre_dispatch
    self.training_type_plugin.pre_dispatch()
pytorch_lightning/plugins/training_type/deepspeed.py:241: in pre_dispatch
    self.init_deepspeed()
pytorch_lightning/plugins/training_type/deepspeed.py:258: in init_deepspeed
    self._initialize_deepspeed_train(model)
pytorch_lightning/plugins/training_type/deepspeed.py:287: in _initialize_deepspeed_train
    model, optimizer, _, lr_scheduler = deepspeed.initialize(
/usr/local/lib/python3.8/dist-packages/deepspeed/__init__.py:120: in initialize
    engine = DeepSpeedEngine(args=args,
/usr/local/lib/python3.8/dist-packages/deepspeed/runtime/engine.py:149: in __init__
    self._configure_distributed_model(model)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = DeepSpeedEngine(
  (module): LightningDeepSpeedModule(
    (module): ModelParallelBoringModel(
      (layer): Linear(in_features=32, out_features=2, bias=True)
      (linear): Linear(in_features=32, out_features=2, bias=True)
    )
  )
)
model = LightningDeepSpeedModule(
  (module): ModelParallelBoringModel(
    (layer): Linear(in_features=32, out_features=2, bias=True)
    (linear): Linear(in_features=32, out_features=2, bias=True)
  )
)

    def _configure_distributed_model(self, model):
        self.module = model
        if self.fp16_enabled():
            if self.zero_optimization_partition_weights() and any(
                [hasattr(param,
                         'ds_id') for param in self.module.parameters()]):
>               assert all([param.dtype == torch.half for param in self.module.parameters()]), f"Model must initialized in fp16 mode for ZeRO Stage 3."
E               AssertionError: Model must initialized in fp16 mode for ZeRO Stage 3.

/usr/local/lib/python3.8/dist-packages/deepspeed/runtime/engine.py:569: AssertionError

```

File: dockers/base-cuda/Dockerfile [hunk 2]
```diff
@@ -119,7 +119,7 @@ RUN \
 RUN \
     # install DeepSpeed
     # TODO(@SeanNaren): 0.3.15 is broken - skipping to unblock
-    pip install 'deepspeed>=0.3.14,!=0.3.15'
+    pip install 'deepspeed==0.3.14
```
Comment (carmocca) [inline dockers/base-cuda/Dockerfile:122]:
```suggestion
    pip install deepspeed==0.3.14
```

File: dockers/base-cuda/Dockerfile [hunk 3]
```diff
@@ -119,7 +119,7 @@ RUN \
 RUN \
     # install DeepSpeed
     # TODO(@SeanNaren): 0.3.15 is broken - skipping to unblock
```
Comment (carmocca) [inline dockers/base-cuda/Dockerfile:121]:
```suggestion
    # TODO(@SeanNaren): >=0.3.15 is broken - skipping to unblock
```

Comment (awaelchli) [reply dockers/base-cuda/Dockerfile:121]: I will try it but I doubt that it's a deepspeed problem. The comment is probably referring to us having a bug with the new deepspeed version, not the other way around.

Comment (carmocca) [reply dockers/base-cuda/Dockerfile:121]: yes - just proposed this to keep the comment up to date

```suggestion
    # TODO(@SeanNaren): CI failing with `>=0.3.15` - skipping to unblock
```

File: .github/workflows/events-nightly.yml [hunk 4]
```diff
@@ -3,6 +3,7 @@ name: Nightly events
 # https://jasonet.co/posts/scheduled-actions/
 # https://github.community/t/distinct-job-for-each-schedule/17811/2
 on:
+  push: {}
   schedule:
```
Comment (awaelchli) [inline .github/workflows/events-nightly.yml:7]:
```suggestion
on:
  schedule:
```

**Review by kaushikb11 — APPROVED** [review]

**Review by ananthsub — APPROVED** [review]

**Review by ethanwharris — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Pull request merged**